llvm/lib/Target/DirectX/DirectXTargetMachine.cpp

```diff
@@ -79,8 +79,8 @@ class DirectXPassConfig : public TargetPassConfig {
   void addCodeGenPrepare() override {
     addPass(createDXILIntrinsicExpansionLegacyPass());
     addPass(createDXILOpLoweringLegacyPass());
-    addPass(createDXILPrepareModulePass());
     addPass(createDXILTranslateMetadataPass());
+    addPass(createDXILPrepareModulePass());
```

Comment:
Why does the order need to be changed here?

Reply:
Because DXILTranslateMetadataPass needs to use attribute like hlsl.numthreads.
If DXILPrepareModulePass before DXILTranslateMetadataPass. The attribute will be removed.